### PR TITLE
Update to daos major version 1, for libdaos API update

### PR DESCRIPTION
**Pull request by kccain**
And update daos_major definition in spec file to update the "virtual"
Provides (e.g., mpifileutils-mpich-daos-1 instead of
mpifileutils-mpich-daos-0).

PR-repos: daos@PR-3935:53 mpich@PR-44

Signed-off-by: Kenneth Cain <kenneth.c.cain@intel.com>

### CODE REVIEW

File: mpifileutils.spec [hunk 1]
```diff
@@ -197,6 +198,9 @@ done
 %endif
 
 %changelog
+* Thu Dec 17 2020 Kenneth Cain <kenneth.c.cain@intel.com> - 0.10.1-5
```
Comment (daltonbohning) [inline mpifileutils.spec:201]:
```suggestion
* Thu Dec 17 2020 Kenneth Cain <kenneth.c.cain@intel.com> - 0.10.1-6
```

File: mpifileutils.spec [hunk 2]
```diff
@@ -43,7 +44,7 @@
 
 Name:		mpifileutils
 Version:	0.10.1
-Release:	4%{?relval}%{?dist}
+Release:	5%{?relval}%{?dist}
```
Comment (daltonbohning) [inline mpifileutils.spec:47]:
```suggestion
Release:	6%{?relval}%{?dist}
```

**Review by daltonbohning — COMMENTED** [review]
Once #6 lands (probably Monday), we'll want to merge this PR with master and bump the release to `6`

**Comment by daltonbohning** [comment]
@kccain It looks like the packaging has been updated to resolve the intermittent Leap repo issues.
It would be good to update the packaging in this PR.
For reference, this is how:
1. Clone the daos-stack/packaging repo to somewhere sensible:
```
git clone git@github.com:daos-stack/packaging.git
```
2. `cd` to your `mpifileutils-pkg` directory and branch (this PR)
3. Run `install` from daos-stack/packaging (the repo you just cloned):
```
/path/to/packaging install
```
Then, for this PR, you should see `modified:   packaging/rpm_chrootbuild`

**Comment by kccain** [comment]
> @kccain It looks like the packaging has been updated to resolve the intermittent Leap repo issues.
> It would be good to update the packaging in this PR.
> For reference, this is how:
> 
> 1. Clone the daos-stack/packaging repo to somewhere sensible:
> 
> ```
> git clone git@github.com:daos-stack/packaging.git
> ```
> 
> 1. `cd` to your `mpifileutils-pkg` directory and branch (this PR)
> 2. Run `install` from daos-stack/packaging (the repo you just cloned):
> 
> ```
> /path/to/packaging install
> ```
> 
> Then, for this PR, you should see `modified: packaging/rpm_chrootbuild`

I have already updated the rpm_chrootbuild in this PR, by using the contents from the daos-stack/daos repositor, utils/rpms/packaging folder. This was a suggested approach while waiting for the packaging repo itself to be updated, which it has as you noted. In my local packaging repo I did git remote update, pull origin master. Then in my mpifileutils-pkg local repo I ran the /path/to/packaging/install script and it produced no modified files. As far as I can tell this change included in this PR is what is needed to address the intermittent leap repo issues.
https://github.com/daos-stack/mpifileutils-pkg/pull/7/files#diff-2263108892edc329d7def9abdf776218a20ddad51d58c5831ec61eb7e2418768

**Comment by daltonbohning** [comment]
> I have already updated the rpm_chrootbuild in this PR, by using the contents from the daos-stack/daos repositor, utils/rpms/packaging folder. This was a suggested approach while waiting for the packaging repo itself to be updated, which it has as you noted. In my local packaging repo I did git remote update, pull origin master. Then in my mpifileutils-pkg local repo I ran the /path/to/packaging/install script and it produced no modified files. As far as I can tell this change included in this PR is what is needed to address the intermittent leap repo issues.
> https://github.com/daos-stack/mpifileutils-pkg/pull/7/files#diff-2263108892edc329d7def9abdf776218a20ddad51d58c5831ec61eb7e2418768

Oh, sorry about that. I guess the Linter must be complaining about something else then, but I'm not sure what exactly:
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fmpifileutils-pkg/detail/PR-7/7/pipeline/27

We don't have "--repo" in the Makefile, and I thought the "No such file or directory" errors were benign. Maybe @brianjmurrell has some insight here?

For what it's worth, I restarted the "master" build, since it failed due to the Leap repo issue, and it also has the same linter issue, although #6 didn't report any issues.

**Comment by kccain** [comment]
> Oh, sorry about that. I guess the Linter must be complaining about something else then, but I'm not sure what exactly:
> https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fmpifileutils-pkg/detail/PR-7/7/pipeline/27
>
I did get an email during the PR build warning that the packaging was out of date. I am guessing this may just be due to a simple difference with the packaging repo contents. In this case though the mpifileutils-pkg packaging/ dir was updated with something newere, not yet landed to packaging. Rather than the usual situation having older content.

**Review by brianjmurrell — APPROVED** [review]

**Comment by brianjmurrell** [comment]
The pending PR causing:

image

has landed, so I did a rebuild to get [build number 8](https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fmpifileutils-pkg/detail/PR-7/8/pipeline) which should be successful, and can be referenced as `mpifileutils-pkg@PR-7` without any `:<build_number>` postfix.

**Review by daltonbohning — APPROVED** [review]

**Comment by kccain** [comment]
This PR appears ready to land based on testing:

1) daos PR-3935 build 54 (provides libdaos.so.1 that this package calls)
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-3935/54/pipeline/

2) daos master (PR-4292) build 8 (provides libdaos.so.0 that this package does not call) - test results show that this package was not installed and therefore did not break current daos master testing
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-3935/54/pipeline/

**Pull request merged**